tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ImmutableMetricName.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.tritium.metrics.registry;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import com.palantir.logsafe.Preconditions;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** ABI backcompat stub for code which was previously generated with immutables. */
```

Comment:
Should we add an explicit comment about why we're not using Immutables  for creation & allocation sensitivity?